### PR TITLE
Non-record: Stacked hyperparameter tuning + eval2048 (RTX 5090, val_bpb 1.336)

### PR DESCRIPTION
**Pull request by gwelinder**
## Non-record submission: Stacked Hyperparameter Tuning + Eval2048

**val_bpb: 1.3358** (post-quant int8+zlib) | **15.8MB artifact** | RTX 5090, 20 train shards

### What this is

40+ automated experiments via an autoresearch loop on the baseline 9x512 architecture. No architecture changes. 5 stacked config fixes improve val_bpb by 0.027.

### Key finding

`WARMDOWN_ITERS=1200` is broken at 600s wallclock. At ~620ms/step you get ~968 total steps, so 1200 > total steps and the cosine warmdown fires from step 1. Fix: `WARMDOWN_ITERS=3000`. (PRs #48 and #73 flagged the same thing.)

### Stacked config
```
WARMDOWN_ITERS=3000, MATRIX_LR=0.06, LOGIT_SOFTCAP=15, MUON_MOMENTUM=0.99
TRAIN_BATCH_TOKENS=131072 (quarter-batch), EVAL_SEQ_LEN=2048
```

### Negative results (also in the README)
- Butterfly/Monarch MLP: 7MB artifact but 1.46 bpb
- Reservoir random MLPs: 2.14 bpb
- Depth recurrence (4x3=12 eff layers): 1.50 bpb
- 6 alternative shapes: none beat 9x512

### Also in `train_gpt.py`
- EVAL_SEQ_LEN decoupling (train short, eval long)
- Alias-aware serialization (shared weights stored once)
- Mixed int6/int8 quantization (INT6_ALL_BLOCK_MATRICES env var)
- Sliding-window eval (EVAL_STRIDE env var, batched)
- Depth recurrence support (NUM_UNIQUE_LAYERS, NUM_RECURRENCE)

Hardware: RTX 5090, RunPod. Not 8xH100. This is a dev iteration result with interesting negative findings.

### CODE REVIEW

**Comment by MatoTeziTanka** [comment]
### Community Review — Non-record: Stacked hyperparameter tuning + eval2048 (RTX 5090, val_bpb 1.336)

**BPB:** 1.336 | **Compliance:** LOOKS CLEAN — pure-neural submission, no TTT/SLOT/n-gram-cache

**What I found in the code** (head SHA `d154c86cfd86`, file `records/track_non_record_16mb/2026-03-19_StackedHyperparams_SP1024_9x512_KV4_RTX5090_gwelinder/train_gpt.py`):

Static code review found no TTT adaptation function, no SLOT optimization loop, no n-gram-cache class, and no pre-quant val-token fine-tune. The eval path uses the standard sliding-window stride-64 pattern. The submission is a pure-neural architecture iteration on the standard SP1024/SP4096/SP8192 baseline.

**CPU smoke test (CT2038 proteus-engine, 2026-04-11):** import OK in 0.03s, dim=512, layers=9, vocab=1024, code=62873 B, SMOKE_TEST_PASS

**Verdict:** LOOKS CLEAN.

**Recommendation to @cocohearts @valerio-oai @0hq @yuzhougu-oai @notapplica:** **MERGE** pending the usual record-track checks (3-seed validation, under-16MB artifact cap, ≤600s train + ≤600s eval on 8×H100 SXM). No compliance flags from the classification pass — this looks like a clean pure-neural iteration on the standard baseline.

*Auto-classification caveat:* this review was drafted by the AST-based classifier. If there's a non-standard eval mechanism (logit postprocessing, hedge mixing, etc.) that I missed because it's factored into a helper file or a non-standard function name, please flag it and I'll re-run the audit manually.

---
*Reviewed by [@MatoTeziTanka](https://github.com/MatoTeziTanka) — [The Agora](https://matotezitanka.github.io/parameter-golf). CPU smoke test (CT2038 proteus-engine, 2026-04-11): import OK in 0.03s, dim=512, layers=9, vocab=1024, code=62873 B, SMOKE_TEST_PASS. Classification via deterministic AST-based `classify_prs.py` (pattern bank derived from ~65 manually-reviewed PRs earlier in the 2026-04-11 sweep). This review was auto-drafted from a template and spot-checked before posting — if the template misread your code, please call it out so I can iterate the classifier.*
